modules/light-clients/07-tendermint/misbehaviour_handle.go

```diff
@@ -53,6 +53,29 @@ func (cs ClientState) CheckForMisbehaviour(ctx sdk.Context, cdc codec.BinaryCode
 	case *Misbehaviour:
 		// The correctness of Misbehaviour ClientMessage types is ensured by calling VerifyClientMessage prior to this function
 		// Thus, here we can return true, as ClientMessage is of type Misbehaviour
+		// if heights are equal check that this is valid misbehaviour of a fork
+		// otherwise if heights are unequal check that this is valid misbehavior of BFT time violation
+		if msg.Header1.GetHeight().EQ(msg.Header2.GetHeight()) {
+			blockID1, err := tmtypes.BlockIDFromProto(&msg.Header1.SignedHeader.Commit.BlockID)
+			if err != nil {
+				return true
+			}
+
+			blockID2, err := tmtypes.BlockIDFromProto(&msg.Header2.SignedHeader.Commit.BlockID)
+			if err != nil {
+				return true
+			}
+
+			// Ensure that Commit Hashes are different
+			if bytes.Equal(blockID1.Hash, blockID2.Hash) {
+				return true
+			}
+
+		} else if msg.Header1.SignedHeader.Header.Time.After(msg.Header2.SignedHeader.Header.Time) {
+			// Header1 is at greater height than Header2, therefore Header1 time must be less than or equal to
+			// Header2 time in order to be valid misbehaviour (violation of monotonic time).
+			return true
```

Comment:
```suggestion
		} else if !msg.Header1.SignedHeader.Header.Time.After(msg.Header2.SignedHeader.Header.Time) {
			// Header1 is at greater height than Header2, therefore Header1 time must be less than or equal to
			// Header2 time in order to be valid misbehaviour (violation of monotonic time).
			return true
```

[`ValidateBasic` ensures that header1 height is greater than or equal to header2 height](https://github.com/cosmos/ibc-go/blob/main/modules/light-clients/07-tendermint/misbehaviour.go#L86-L89). Therefore, header1 time must be after header2. If it is not, then we violate time monotonicity and should return true as this is considered misbehaviour.